hydromt/drivers/pandas_driver.py

```diff
@@ -0,0 +1,87 @@
+"""Driver for DataFrames using the pandas library."""
+from logging import Logger, getLogger
+from pathlib import Path
+from typing import List, Optional
+
+import pandas as pd
+
+from hydromt._typing import NoDataStrategy, StrPath, TimeRange, Variables
+from hydromt._utils.unused_kwargs import warn_on_unused_kwargs
+from hydromt.drivers import DataFrameDriver
+
+logger: Logger = getLogger(__name__)
+
+
+class PandasDriver(DataFrameDriver):
+    """Driver for DataFrames using the pandas library."""
+
+    name = "pandas"
+
+    def read_data(
+        self,
+        uris: List[str],
+        *,
+        variables: Optional[Variables] = None,
+        time_range: Optional[TimeRange] = None,
+        logger: Logger = logger,
+        handle_nodata: NoDataStrategy = NoDataStrategy.RAISE,
+    ) -> pd.DataFrame:
+        """Read in any compatible data source to a pandas `DataFrame`."""
+        if len(uris) > 1:
+            raise ValueError(
+                "DataFrame: Reading multiple files with the "
+                f"{self.__class__.__name__} driver is not supported."
+            )
+        warn_on_unused_kwargs(
+            self.__class__.__name__, {"time_range": time_range}, logger
+        )
+        uri = uris[0]
+        extension: str = uri.split(".")[-1]
+        logger.info(f"Reading using {self.name} driver from {uri}")
+        if extension == "csv":
+            return pd.read_csv(uri, usecols=variables, **self.options)
+        elif extension == "parquet":
+            return pd.read_parquet(uri, columns=variables, **self.options)
+        elif extension in ["xls", "xlsx"]:
+            return pd.read_excel(
+                uri, usecols=variables, engine="openpyxl", **self.options
+            )
+        elif extension in ["fwf"]:
+            warn_on_unused_kwargs(
+                self.__class__.__name__, {"variables": variables}, logger
+            )
+            return pd.read_fwf(uri, **self.options)
```

Comment:
seems like this implementation can read `fwf` but not write it. Is that correct? I'm not familiar with the format. Do we need it at all?

Reply:
Yes, pandas does not write to fwf: https://pandas.pydata.org/docs/user_guide/io.html